rita/src/rita_client/dashboard/network_endpoints.rs

```diff
@@ -54,24 +58,102 @@ pub fn get_exit_info(_req: HttpRequest) -> Box<Future<Item = Json<Vec<ExitInfo>>
-    debug!("/exits/{}/reset hit", name);
+pub fn reset_exit(path: Path<String>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    let exit_name = path.into_inner();
+    debug!("/exits/{}/reset hit", exit_name);
+
+    let mut exits = SETTING.get_exits_mut();
+    let mut ret = HashMap::new();
+
+    if let Some(exit) = exits.get_mut(&exit_name) {
+        info!("Changing exit {:?} state to New", exit_name);
+        exit.info = ExitState::New;
+        return Box::new(future::ok(HttpResponse::Ok().json(ret)));
+    } else {
+        error!("Requested a reset on unknown exit {:?}", exit_name);
+        ret.insert(
+            "error".to_owned(),
+            format!("Requested reset on unknown exit {:?}", exit_name),
+        );
+        return Box::new(future::ok(
+            HttpResponse::new(StatusCode::BAD_REQUEST)
+                .into_builder()
+                .json(ret),
+        ));
+    }
+}
 
-    Dashboard::from_registry()
-        .send(ResetExit(name.into_inner()))
-        .from_err()
-        .and_then(move |reply| Ok(Json(reply?)))
-        .responder()
+pub fn select_exit(path: Path<String>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    let exit_name = path.into_inner();
+    debug!("/exits/{}/select hit", exit_name);
+
+    let mut exit_client = SETTING.get_exit_client_mut();
+    let mut ret = HashMap::new();
+
+    if exit_client.exits.contains_key(&exit_name) {
+        info!("Selecting exit {:?}", exit_name);
+        exit_client.current_exit = Some(exit_name);
+        return Box::new(future::ok(HttpResponse::Ok().json(ret)));
+    } else {
+        error!("Requested selection of an unknown exit {:?}", exit_name);
+        ret.insert(
+            "error".to_owned(),
+            format!("Requested selection of an unknown exit {:?}", exit_name),
+        );
+        return Box::new(future::ok(
+            HttpResponse::new(StatusCode::BAD_REQUEST)
+                .into_builder()
+                .json(ret),
+        ));
+    }
 }
 
-pub fn select_exit(name: Path<String>) -> Box<Future<Item = Json<()>, Error = Error>> {
-    debug!("/exits/{}/select hit", name);
+pub fn register_to_exit(path: Path<String>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    let exit_name = path.into_inner();
+    debug!("/exits/{}/register hit", exit_name);
+
+    debug!("Attempting to register on exit {:?}", exit_name);
+
+    Box::new(exit_setup_request(exit_name, None).then(|res| {
+        let mut ret = HashMap::new();
+        match res {
+            Ok(_) => future::ok(HttpResponse::Ok().json(ret)),
+            Err(e) => {
+                error!("exit_setup_request() failed with: {:?}", e);
+                ret.insert("error".to_owned(), "Exit setup request failed".to_owned());
+                ret.insert("rust_error".to_owned(), format!("{:?}", e));
+                future::ok(
+                    HttpResponse::new(StatusCode::BAD_REQUEST)
```

Comment:
Woudln't server error (500+) code fit more here? `BAD_REQUEST` is a client error which would imply bad user input or something similiar, and we clearly handle an error here.

Reply:
Yeah but problem is exit_setup_request only checks if the exit exists inside, so I have no way of knowing from the outside, and checking again outside is kinda janky.